tools/wptserve/wptserve/wptserve.py

```diff
@@ -24,10 +24,10 @@ def parse_args():
 
 def main():
     args = parse_args()
-    httpd = server.WebTestHttpd(host=args.host, port=args.port,
-                                use_ssl=False, certificate=None,
-                                doc_root=args.document_root)
+    httpd = WebTestHttpd(host=args.host, port=args.port,
+                         use_ssl=False, certificate=None,
+                         doc_root=args.document_root)
     httpd.start()
 
 if __name__ == "__main__":
-    main()
+    main()  # type: ignore
```

Comment:
Does this cause problems if the type is not explicitly ignored? `main` should not return any value so it could be `# type: None` or possibly change  `main` to definitively return `None` (i.e. `def main() -> None:` )

Reply:
Adding `def main() -> None` will result in these errors in turn:

```
tools/wptserve/wptserve/wptserve.py:26: error: Call to untyped function "parse_args" in typed context  [no-untyped-call]
tools/wptserve/wptserve/wptserve.py:27: error: Call to untyped function "WebTestHttpd" in typed context  [no-untyped-call]
tools/wptserve/wptserve/wptserve.py:30: error: Call to untyped function "start" in typed context  [no-untyped-call]
```

One really has to annotate from the bottom up, not from the top down. The problem is that the top-level context is typed, so calls like these one need to be decorated to avoid going full top-down from the beginning.